scipy/ndimage/tests/test_ndimage.py

```diff
@@ -2265,6 +2265,13 @@ def test_rotate08(self):
             out = ndimage.rotate(data, 90, axes=(0, 1), reshape=False)
             assert_array_almost_equal(out, expected)
 
+    def test_rotate09(self):
```

Comment:
I think a bit of a more nontrivial image with a nontrivial angle needs to be tested to be sure. This is a bit too simplistic for the change involved.

Reply:
The test is about asserting correct error raise if any axis is equal to `input.ndim`. That is the original bug reported in #10011.
But I can add an other test asserting that the refactoring doesn't introduce any regression.